ts/utils/fetch.ts

```diff
@@ -114,3 +129,246 @@ export const constantPollingFetch = (
 
   return retriableFetch(retryWithTransient404s, shouldAbort)(timeoutFetch);
 };
+
+/**
+ * Decorates the current fetch with LolliPOP headers and http-signature
+ */
+export function lollipopFetch(
+  timeout: Millisecond = fetchTimeout,
+  maxRetries: number = fetchMaxRetries,
+  lollipopConfig: LollipopConfig,
+  keyInfo: KeyInfo
+) {
+  const fetchApi = (global as any).fetch;
+  const abortableFetch = AbortableFetch(fetchApi);
+  const timeoutFetch = toFetch(
+    setFetchTimeout(timeout, abortableFetch)
+  ) as FixedFetch;
+  // configure retry logic with default exponential backoff
+  // @see https://github.com/pagopa/io-ts-commons/blob/master/src/backoff.ts
+  const exponentialBackoff = calculateExponentialBackoffInterval();
+  const retryLogic = withRetries<Error, Response>(
+    maxRetries,
+    exponentialBackoff
+  );
+  const retryWithTransient429s = retryLogicForTransientResponseError(
+    _ => _.status === 429,
+    retryLogic
+  );
+
+  return retriableFetch(retryWithTransient429s)(
```

Comment:
This code seems too long. 
Is it possible to split it into further methods and maybe add some comments?